packages/backend-core/src/context/mainContext.ts

```diff
@@ -281,6 +281,27 @@ export function doInScimContext(task: any) {
   return newContext(updates, task)
 }
 
+export async function ensureSnippetContext() {
+  const ctx = getCurrentContext()
+
+  // If we've already added snippets to context, continue
+  if (!ctx || ctx.snippets) {
+    return
+  }
+
+  // Otherwise get snippets for this app and update context
+  let snippets: Snippet[] | undefined
+  const db = getAppDB()
+  if (db) {
+    const app = await db.get<App>(DocumentType.APP_METADATA)
+    snippets = app.snippets
+  }
+
+  // Always set snippets to a non-null value so that we can tell we've attempted
+  // to load snippets
+  ctx.snippets = snippets || []
+}
```

Comment:
Snippet context works similarly to how we cache the VM isolate - it mutates the current context rather than creating a new layer. The main motivation for this is that there are a large number of places where things like formula columns need to be processed, and attempting to wrap all of those in a new context layer to add snippets while avoiding fetching the snippet definitions multiple times would be extremely difficult.

Doing it this way makes it trivial to ensure that we're only every fetch snippet definitions once per action / API call / automation, as we can simply call `ensureSnippetContext` where needed and not worry about doubling up on app DB reads.

I appreciate this is different than the normal approach though, so I'm open to feedback.